src/main/java/com/grepp/carrierroute/booking/service/HotelBookingService.java

```diff
@@ -0,0 +1,88 @@
+package com.grepp.carrierroute.booking.service;
+
+import com.grepp.carrierroute.booking.domain.HotelBooking;
+import com.grepp.carrierroute.booking.dto.HotelBookingRequestDto;
+import com.grepp.carrierroute.booking.dto.HotelBookingResponseDto;
+import com.grepp.carrierroute.booking.exception.InsufficentRoomException;
+import com.grepp.carrierroute.booking.repository.HotelBookingRepository;
+import com.grepp.carrierroute.booking.service.converter.HotelBookingConverter;
+import com.grepp.carrierroute.hotel.domain.HotelRoom;
+import com.grepp.carrierroute.hotel.exception.ErrorMessage;
+import com.grepp.carrierroute.hotel.repository.HotelRoomRepository;
+import com.grepp.carrierroute.user.domain.User;
+import com.grepp.carrierroute.user.exception.UserNotFoundException;
+import com.grepp.carrierroute.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class HotelBookingService {
+    private final UserRepository userRepository;
+    private final HotelRoomRepository hotelRoomRepository;
+    private final HotelBookingRepository hotelBookingRepository;
+    private final HotelBookingConverter converter;
+
+    public HotelBookingResponseDto bookRooms(HotelBookingRequestDto bookingRequestDto, String userId){
+        User user = findUserBy(userId);
+        List<HotelRoom> roomsToBook = getAvailableRooms(bookingRequestDto);
+        long totalPrice = calculateTotalPrice(roomsToBook, bookingRequestDto.getCheckInDate(), bookingRequestDto.getCheckOutDate());
+
+        List<HotelBooking> bookings = bookRooms(user, roomsToBook, totalPrice, bookingRequestDto);
+
+        return converter.convertToHotelBookingResponseDto(totalPrice, bookings, bookingRequestDto);
+    }
+
+    private User findUserBy(String userId){
+        return userRepository.findById(userId)
+                .orElseThrow(() -> new UserNotFoundException(ErrorMessage.USER_NOT_FOUNDED.getMessage()));
+    }
```

Comment:
```suggestion
    private User findUserByUserId(String userId){
        return userRepository.findById(userId)
                .orElseThrow(() -> new UserNotFoundException(ErrorMessage.USER_NOT_FOUNDED.getMessage()));
    }
```

By 뒤에 뭘로 검색하는지 있으면 좋을 것 같아요~

Reply:
> By 뒤에 뭘로 검색하는지 있으면 좋을 것 같아요~

파라미터가 추가되면 By~~ 뒤에 뭘 이용할 것인지 계속 추가되니까 굳이 뭘 이용하는지 작성할 필요가 있을까 싶어서,
저는 일괄적으로 메소드 명을 By까지만 작성했는데, 어떻게 생각하세요~?

Reply:
오홍 그럴 수도 있겠네요. 이 부분도 레이한테 물어보는 걸로 하시죠 ㅎ

Reply:
하 이거 참 난제네요,, 저도 매개변수명을 통해 설명을 부가한다는 측면에서 괜찮다고 생각했는데
findUserBy(String userId) 도 만들어야 되고 findUserBy(String name) 도 만들어야 하는 상황이 발생한다면
오버로딩이 불가하다는 측면에서 메서드 명에 언급되어지는게 더 나은 것 같습니다!?ㅎㅎ

Reply:
사실 이 메소드 명을 findBy~~ 이런 식으로 해서 생기는 문제 같네요!
service에서는 findBy~ 이것보다는 getUser로 하는 것이 더 명확한 메소드 명인 것 같아서 `getUser`로 수정했습니다!

HotelService 코드에서도 이런 식으로 작성했는데 추후에 일관성 있게 수정해야겠네요! 여러 의견 감사해요! 울팀 짱>_<